File: easy/Self-Describing-Numbers/Self Describing Numbers.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            // Process line of input Here
            ShowLongestWord(line);
        }
    }
    
    public static void ShowLongestWord(String line){
        String[] digs = line.split("");
        boolean selfname = true;
        int i = 0;
        while (i<digs.length && selfname){
            int j = 0;
            int cont = 0;
            while(j<digs.length && selfname){
                if(i==Integer.valueOf(digs[j]))cont++; 
                j++;
            }
            if (Integer.valueOf(digs[i]) != cont) selfname = false;
            i++;
        }
        if(selfname) System.out.println("1");
        else System.out.println("0");
    }
}

File: easy/Self-Describing-Numbers/Self Describing Numbers.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            ShowLongestWord(line);
        }
    }
    
    public static void ShowLongestWord(string line){
        bool selfname = true;
        int i = 0;
        while (i<line.Length && selfname){
            int j = 0;
            int cont = 0;
            while(j<line.Length && selfname){
                if(i==Convert.ToInt32(line[j].ToString()))cont++; 
                j++;
            }
            if (Convert.ToInt32(line[i].ToString()) != cont) selfname = false;
            i++;
        }
        Console.WriteLine(selfname?"1":"0");
    }
}
